cf-custom-resources/test/cert-replicator-test.js

```diff
@@ -0,0 +1,528 @@
+          TargetRegion: "us-east-1",
+          EnvRegion: "us-west-2",
+          CertificateArn: testCertificateArn,
+        },
+        OldResourceProperties: {},
+      })
+      .expectResolve(() => {
+        sinon.assert.calledWith(
+          describeCertificateFake,
+          sinon.match({
+            CertificateArn: testCertificateArn,
+          })
+        );
+        expect(request.isDone()).toBe(true);
+      });
+  });
+
+  test("Create operation fails if cannot request new certificate", () => {
+    const requestCertificateFake = sinon.fake.rejects(new Error("some error"));
+
+    const describeCertificateFake = sinon.stub();
+    describeCertificateFake.resolves({
+      Certificate: {
+        CertificateArn: testCertificateArn,
+        DomainName: testDomainName,
+        SubjectAlternativeNames: testSANs,
+      },
+    });
+
+    AWS.mock("ACM", "requestCertificate", requestCertificateFake);
+    AWS.mock("ACM", "describeCertificate", describeCertificateFake);
+
+    const request = nock(ResponseURL)
+      .put("/", (body) => {
+        return body.Status === "FAILED";
+      })
+      .reply(200);
+
+    return LambdaTester(handler.certificateReplicateHandler)
+      .event({
+        RequestType: "Create",
+        RequestId: testRequestId,
+        ResourceProperties: {
+          AppName: testAppName,
+          EnvName: testEnvName,
+          TargetRegion: "us-east-1",
+          EnvRegion: "us-west-2",
+          CertificateArn: testCertificateArn,
+        },
+        OldResourceProperties: {},
+      })
+      .expectResolve(() => {
+        sinon.assert.calledWith(
+          describeCertificateFake,
+          sinon.match({
+            CertificateArn: testCertificateArn,
+          })
+        );
+        sinon.assert.calledWith(
+          requestCertificateFake,
+          sinon.match({
+            DomainName: testDomainName,
+            SubjectAlternativeNames: testSANs,
+            ValidationMethod: "DNS",
+            Tags: testCopilotTags,
+          })
+        );
+        expect(request.isDone()).toBe(true);
+      });
+  });
+
+  test("Update operation requests a new certificate", () => {
+    const requestCertificateFake = sinon.fake.resolves({
+      CertificateArn: testCertificateArn,
+    });
+
+    const describeCertificateFake = sinon.stub();
+    describeCertificateFake.resolves({
+      Certificate: {
+        CertificateArn: testCertificateArn,
+        DomainName: testDomainName,
+        SubjectAlternativeNames: testSANs,
+      },
+    });
+
+    const waitForCertificateValidationFake = sinon.stub();
+    waitForCertificateValidationFake.resolves();
+
+    AWS.mock("ACM", "requestCertificate", requestCertificateFake);
+    AWS.mock("ACM", "describeCertificate", describeCertificateFake);
+    AWS.mock("ACM", "waitFor", waitForCertificateValidationFake);
+
+    const request = nock(ResponseURL)
+      .put("/", (body) => {
+        return body.Status === "SUCCESS";
```

Comment:
Similarly, what do you think of matching the `PhysicalResourceID` here as well, that it is indeed the requested cert ARN?

Reply:
🥺 yes please, I'd be great to have a testcase that ensures this invariant: https://github.com/aws/copilot-cli/pull/3837/files#r936022666